ouroboros-consensus/docs/internal-CHANGELOG.md

```diff
@@ -0,0 +1,83 @@
+# Internal Change Log
+
+  - This change log is primarily intended to benefit downstream Cardano
+    development teams. We therefore are not commiting to discuss changes that
+    aren't relevant to them -- eg those changes would likely be most useful to
+    list in the `cardano-node` release notes.
+
+  - So far, versioning of "internal" Cardano libraries is based only on commit
+    hashes, except for the `cardano-node` repo.
+
+For those reasons, we primarily organize this log chronologically: most recent
+changes first.
+
+In order to identify changes by commit hash, Pull Request number, precise
+timestamp, etc, invoke `git blame --first-parent` or `git blame` on this file.
+
+Relatedly, we will NEVER alter entries in this log (until we shift to an
```

Comment:
Is it a problem to fix up or correct old changelog entries? It's always nice to get it right the first time, in the relevant PR, but I don't think anyone would mind previous entries being mutated.

I plan to read the changelog by running `git diff revA revB --  ouroboros-consensus/docs/internal-CHANGELOG.md`.

Reply:
I guess the intention is to keep the alignment between `git blame` and the original commit?

Reply:
@nc6 that's my understanding, but as i mention [below](https://github.com/input-output-hk/ouroboros-network/pull/3279#discussion_r679113587) I think we can weaken this requirement to simply the date lines being immutable.

Reply:
Thanks for the sharing the explicit command, Rodney. I found that very helpful.

Clarke: yes.

David: I'd say you're ultimately right about the dateline being the only necessarily immutable one, but I still think there is value in treating the entries as append-only.

Reply:
@nfrisby absolutely agree